Lib/unittest/loader.py

```diff
@@ -425,11 +425,20 @@ def _find_test_path(self, full_path, pattern, namespace=False):
         """
         basename = os.path.basename(full_path)
         if os.path.isfile(full_path):
-            if not VALID_MODULE_NAME.match(basename):
-                # valid Python identifiers only
-                return None, False
+            root, ext = os.path.splitext(basename)
+            try:
+                if not (ext == '.py' and root.isidentifier()):
+                    # valid Python identifiers only
+                    return None, False
+            except AttributeError:
+                # Python 2 does not have isidentifier() and also has simpler
```

Comment:
Note for core reviewers: Would this be backported to Python 2? I doubt it, in which case this change can be dramatically simplified.